src/main/java/Entities/Group.java

```diff
@@ -3,13 +3,33 @@
 import java.util.HashMap;
 
 public interface Group extends Serializable {
-
+    /**
+     *
+     * @return the group's unique group name
+     */
     public String getGroupName();
-    public HashMap<String, User> getGroupMembers(HashMap<String, User> userMap);
+    /**
+     *
+     * @param userMap a map (key: username, value: User) of all the existing Users in the system
+     * @return a map of the group members' usernames to User objects
+     */
 
+    public HashMap<String, User> getGroupMembers(HashMap<String, User> userMap);
+    /**
+     *
+     * @param userMap a map (key: username, value: User) of all the existing Users in the system
+     * @return a map (key: username, value: User) of the users who applied to be in this group
+     */
     public HashMap<String, User> getMemberRequests(HashMap<String, User> userMap);
 
-    public boolean setGroupLeader(String userID);
+    /**
+     *
+     * @param username of the User who wants to be the new Group Leader
+     * @return whether the Group Leader was successfully changed or not
+     */
+    public boolean setGroupLeader(String username);
+
+    //public GroupProfile getProfile();
 
 
```

Comment:
Hi could you add getGroupMembersUserNames to the interface? My code takes in the interface not Normal Group. Also where is getGroupProfile? Maybe I am missing something. 

It would be helpful that we have all necessary methods for a "Group" to have in the interface, otherwise we will have to make all our use cases dependent on the NormalGroup, and create new ones for any other Group made.

Reply:
Yes, I will include getGroupMembersUserNames in the next PR. I don't have getGroupProfile or setGroupProfile yet as I need Group Profile to be merged first, but will do it as soon as it is.